### PR TITLE
FISH-6013 upgrade jakarta-annnotation to 2.1.0

### PR DESCRIPTION
**Pull request by aubi**
## Description
Upgrade jakarta-annnotation to 2.1.0 as a path to Jakarta EE 10 compatibility

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -147,7 +147,7 @@
         <jakarta.el-api.version>4.0.0</jakarta.el-api.version>
         <javax.cache-api.version>1.1.1.payara-p1</javax.cache-api.version>
         <mail.version>2.0.1</mail.version>
-        <jakarta.annotation-api.version>2.0.0</jakarta.annotation-api.version>
+        <jakarta.annotation-api.version>2.1.0-B1</jakarta.annotation-api.version>
```
Comment (breakponchito) [inline pom.xml:150]: it is correct this version or the 2.1.0?  I saw this last tag: https://github.com/eclipse-ee4j/common-annotations-api/releases/tag/2.1.0

**Comment by Pandrex247** [comment]
Added label to make it obvious that this shouldn't go in Alpha2

**Comment by aubi** [comment]
Jenkins test please

**Comment by aubi** [comment]
broken build fixed by PR https://github.com/payara/Payara/pull/5643

**Pull request merged**